uds/database/abstract_data_record.py

```diff
@@ -0,0 +1,123 @@
+"""
+Definition of all Data Records types.
+
+Each Data Record contains mapping (translation) of raw data (sequence of bits in diagnostic message payload) to some
+meaningful information (e.g. physical value, text).
+"""
+
+__all__ = ["DataRecordType", "AbstractDataRecord", "DecodedDataRecord"]
+
+from abc import ABC, abstractmethod
+from typing import Tuple, TypedDict, Union
+
+from uds.utilities import ValidatedEnum
+
+DataRecordPhysicalValueAlias = Union[int, float, str, Tuple["DecodedDataRecord", ...]]
+"""Alias of Data Records' physical value."""
+
+
+class DecodedDataRecord(TypedDict):
+    """Structure of decoded Data Record."""
```

Comment:
What is a DataRecord? Is it each possible part of a UDS message (request/response), for example, a Service ID, sub-function, NRC, or DID that we read from the ECU? If so, perhaps the DecodedDataRecord should include two additional attributes: "offset" and "factor," which are often needed to decode signals?

Reply:
If you look into standard ISO-14229-1, every request and response table is divided into rows. Each row is a single DataRecord:
![image](https://github.com/user-attachments/assets/111bb9c4-1d4c-4d3d-b99a-f33714428498)

Reply:
Ok i get it, it should be object for each single part of the UDS message.